### ISSUE TITLE
After upgrading to 1.60.5, Api method property "Invoke with caller credentials" cannot be set anymore

### ISSUE THREAD

**Issue by ftmazzone** [serverless/serverless#7194]
# Bug Report

## Description

1. What did you do?

`npx serverless deploy --id test`

1. What happened?

The default generated "resourcePolicy" for ApiGateway is conflicting with the method option "Invoke with caller credentials".

* **cloudFormation error :**

`Caller provided credentials not allowed when resource policy is set (Service: AmazonApiGateway; Status Code: 400; Error Code: BadRequestException`

1. What should've happened?
Deployment should be successful and the API method property "Invoke with caller credentials" should be activated

1. What's the content of your `serverless.yml` file?

```
service: 
  name: serverless-tester-${self:custom.service}

provider:
  name: aws
  runtime: nodejs12.x
  stage:  ${file(./config.${self:custom.id}.json):stage}
  region:  ${file(./config.${self:custom.id}.json):region}
  endpointType: regional
  logRetentionInDays: 30
  versionFunctions: false

  deploymentBucket:
    name: ${file(./config.${self:custom.id}.json):deploymentBucketName}

package:
  excludeDevDependencies: true

functions:
  internalApiTester:
    handler: src/genericLambda.handler
    timeout: 30
    memorySize: 256
    vpc: ${file(./config.${self:custom.id}.json):vpc}
    events:
      - http:
          path: /generic/tester
          method: GET
          authorizer: aws_iam
resources:
  Resources:
    ApiGatewayMethodGenericTesterGet:
      Type: AWS::ApiGateway::Method
      Properties:
        Integration:
          Credentials: "arn:aws:iam::*:user/*"
plugins:
  - serverless-pseudo-parameters
custom:
  id: ${opt:id}
  service: ${file(./config.${self:custom.id}.json):service}
```

This issue seems to be related to the pull request

- #7138

**Comment by medikoo** [comment]
Thanks @ftmazzone for report. Can you update the test case, so it doesn't involve any plugins?

(if the issue is influenced by a plugin, the bug report should be reported at plugin repository)

**Comment by carstenmuellerlemberg** [comment]
In substitution for Florent, here is the example:
----------------------------------------------------------

```
service:

  name: serverless-tester

provider:

  name: aws

  runtime: nodejs12.x

  stage:  dev
  region:  us-east-1
  endpointType: regional
  logRetentionInDays: 30
  versionFunctions: false

package:

  excludeDevDependencies: true

functions:

  internalApiTester:
    handler: src/genericLambda.handler
    timeout: 30

    memorySize: 256

    events:
      - http:
          path: /generic/tester
          method: GET
          authorizer: aws_iam

resources:
  Resources:
    ApiGatewayMethodGenericTesterGet:
      Type: AWS::ApiGateway::Method
      Properties:
        Integration:
          Credentials: "arn:aws:iam::*:user/*"
```

**Comment by medikoo** [comment]
Thank you @carstenmuellerlemberg I can reproduce it now.

I'm in contact with AWS support team, on whether we can address it without a need of reopenning of https://github.com/serverless/serverless/issues/6789 

Anyway it should be fixed shortly

**Comment by ftmazzone** [comment]
Thank you @medikoo for the quick update. To date I did not find any AWS documentation about this limitation.

**Comment by medikoo** [comment]
Will be fixed with #7212
